src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -120,51 +109,45 @@ static InitChainEntry sInitChain[] = {
     ICHAIN_F32(targetArrowOffset, 30, ICHAIN_STOP),
 };
 
-typedef enum {
-    /* 0 */ ENDNS_ANIM_0,
-    /* 1 */ ENDNS_ANIM_1,
-    /* 2 */ ENDNS_ANIM_2
-} EnDnsAnimation;
-
 static AnimationMinimalInfo sAnimationInfo[] = {
     { &gBusinessScrubNervousIdleAnim, ANIMMODE_LOOP, 0.0f },
-    { &gBusinessScrubAnim_4404, ANIMMODE_ONCE, 0.0f },
+    { &gBusinessScrubLeaveBurrowAnim, ANIMMODE_ONCE, 0.0f },
     { &gBusinessScrubNervousTransitionAnim, ANIMMODE_ONCE, 0.0f },
 };
 
 void EnDns_Init(Actor* thisx, PlayState* play) {
     EnDns* this = (EnDns*)thisx;
 
-    if (this->actor.params < 0) {
+    if (EN_DNS_TYPE(this) < 0) {
         // "Function Error (Deku Salesman)"
-        osSyncPrintf(VT_FGCOL(RED) "引数エラー（売りナッツ）[ arg_data = %d ]" VT_RST "\n", this->actor.params);
+        osSyncPrintf(VT_FGCOL(RED) "引数エラー（売りナッツ）[ arg_data = %d ]" VT_RST "\n", EN_DNS_TYPE(this));
         Actor_Kill(&this->actor);
         return;
     }
     // Sell Seeds instead of Arrows if Link is child
-    if ((this->actor.params == 0x0006) && (LINK_AGE_IN_YEARS == YEARS_CHILD)) {
-        this->actor.params = 0x0003;
+    if ((EN_DNS_TYPE(this) == EN_DNS_TYPE_ARROWS) && (LINK_AGE_IN_YEARS == YEARS_CHILD)) {
+        EN_DNS_TYPE(this) = EN_DNS_TYPE_SEEDS;
     }
     // "Deku Salesman"
-    osSyncPrintf(VT_FGCOL(GREEN) "◆◆◆ 売りナッツ『%s』 ◆◆◆" VT_RST "\n", D_809F0424[this->actor.params],
-                 this->actor.params);
+    osSyncPrintf(VT_FGCOL(GREEN) "◆◆◆ 売りナッツ『%s』 ◆◆◆" VT_RST "\n", sItemDebugTxt[EN_DNS_TYPE(this)],
+                 EN_DNS_TYPE(this));
```

Comment:
not your change but
```suggestion
    osSyncPrintf(VT_FGCOL(GREEN) "◆◆◆ 売りナッツ『%s』 ◆◆◆" VT_RST "\n", sItemDebugTxt[EN_DNS_TYPE(this)]);
```
turns out the third argument is fake
(I tried removing it because the format string had no %hx/%hd/whatever for it and it matched)